src/fmeval/eval_algorithms/general_semantic_robustness.py

```diff
@@ -64,51 +55,31 @@
 
 
 @dataclass(frozen=True)
-class GeneralSemanticRobustnessConfig(EvalAlgorithmConfig):
+class GeneralSemanticRobustnessConfig(SemanticRobustnessConfig):
     """Configures the general semantic robustness evaluation algorithm.
 
-    :param perturbation_type: Perturbation type for generating perturbed inputs.
-        Either BUTTER_FINGER, RANDOM_UPPER_CASE, or WHITESPACE_ADD_REMOVE.
-    :param num_perturbations: Number of perturbed outputs to be generated for robustness evaluation.
     :param num_baseline_samples: Only used for non-deterministic models. Number of times we generate
         the model output with the same input to compute the "baseline" change in model output. We
         compute differences between all pairs of outputs, i.e. between comb(num_baseline_samples, 2) pairs.
-    :param butter_finger_perturbation_prob: The probability that a given character will be perturbed.
-        Used when perturbation_type is BUTTER_FINGER.
-    :param random_uppercase_corrupt_proportion: Fraction of characters to be changed to uppercase.
-        Used when perturbation_type is RANDOM_UPPER_CASE.
-    :param whitespace_remove_prob: The probability of removing a whitespace character.
-        Used when perturbation_type is WHITESPACE_ADD_REMOVE.
-    :param whitespace_add_prob: The probability of adding a whitespace character after a non-whitespace character.
-        Used when perturbation_type is WHITESPACE_ADD_REMOVE.
+
     :param model_type_for_bertscore: Model type to use for BERT score.
     """
 
-    perturbation_type: str = BUTTER_FINGER
-    num_perturbations: int = 5
     num_baseline_samples: int = 4
-    butter_finger_perturbation_prob: float = 0.1
-    random_uppercase_corrupt_proportion: float = 0.1
-    whitespace_add_prob: float = 0.05
-    whitespace_remove_prob: float = 0.1
     model_type_for_bertscore: str = BERTSCORE_DEFAULT_MODEL
 
     def __post_init__(self):
-        if self.perturbation_type not in PERTURBATION_TYPE_TO_HELPER_CLASS.keys():
-            raise EvalAlgorithmClientError(
-                f"Invalid perturbation type '{self.perturbation_type} requested, please "
-                f"choose from acceptable values: {PERTURBATION_TYPE_TO_HELPER_CLASS.keys()}"
-            )
-        if not BertscoreModelTypes.model_is_allowed(self.model_type_for_bertscore):
-            raise EvalAlgorithmClientError(
-                f"Invalid model_type_for_bertscore: {self.model_type_for_bertscore} requested in "
-                f"GeneralSemanticRobustnessConfig, please choose from acceptable values: {BertscoreModelTypes.model_list()}."
-            )
-        if self.num_baseline_samples < 2:
-            raise EvalAlgorithmClientError(
-                f"Invalid num_baseline_samples: {self.num_baseline_samples} in GeneralSemanticRobustnessConfig. "
-                f"The value should be at least 2."
-            )
+        super().__post_init__()
+        require(
```

Comment:
Subjective:
I know we follow this style in internal packages. I am not a fan of this though, because it is not readable, requires an extra hop in code to see what kind of exception is being through.

And it is anyways not helping as there is no impact on code modularity or verbiage. 

Hence intentionally we showcased exceptions being thrown.